crates/bevy_scene/src/dynamic_scene.rs

```diff
@@ -138,3 +155,84 @@ where
         .new_line("\n".to_string());
     ron::ser::to_string_pretty(&serialize, pretty_config)
 }
+
+#[cfg(test)]
+mod tests {
+    use bevy_app::AppTypeRegistry;
+    use bevy_ecs::{entity::EntityMap, system::Command, world::World};
+    use bevy_hierarchy::{AddChild, Parent};
+
+    use crate::dynamic_scene_builder::DynamicSceneBuilder;
+
+    #[test]
+    fn components_not_defined_in_scene_should_not_be_effected_by_scene_entity_map() {
+        // Testing that scene reloading applies EntitiyMap correctly to MapEntities components.
+
+        // First, we create a simple world with a parent and a child relationship
+        let mut world = World::new();
+        world.init_resource::<AppTypeRegistry>();
+        world
+            .resource_mut::<AppTypeRegistry>()
+            .write()
+            .register::<Parent>();
+        let gen_0_entity = world.spawn_empty().id();
```

Comment:
Nit: This test might be a bit easier to read if these names were a little more descriptive. Maybe like `world_entity_a`/`scene_entity_a` or something similar?

Reply:
Okay, I did `orginal_`/`from_scene_` prefixes, and then explicitly called them `parent_entity`/`child_entity`.